src/Avalonia.Controls/Control.cs

```diff
@@ -309,5 +331,135 @@ protected override void OnKeyUp(KeyEventArgs e)
+        protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnAttachedToVisualTree(e);
+
+            InvalidateFlowDirection();
+        }
+
+        private void InvalidateFlowDirection()
+        {
+            FlowDirection parentFD = FlowDirection.LeftToRight;
+            FlowDirection thisFD = FlowDirection;
+
+            bool parentShouldGetMirrored = true;
+            bool thisShouldGetMirrored = ShouldGetInvertedIfRightToLeft();
+
+            var parent = this.FindAncestorOfType<Control>();
+            if (parent != null)
+            {
+                parentFD = parent.FlowDirection;
+                parentShouldGetMirrored = parent.ShouldGetInvertedIfRightToLeft();
+            }
+
+            bool shouldBeMirrored = thisFD == FlowDirection.RightToLeft && thisShouldGetMirrored;
+            bool parentMirrored = parentFD == FlowDirection.RightToLeft && parentShouldGetMirrored;
+
+            bool shouldApplyMirrorTransform = shouldBeMirrored != parentMirrored;
+
+            if (shouldApplyMirrorTransform)
+            {
+                AddMirrorTransform();
+            }
+            else
+            {
+                RemoveMirrorTransform();
+            }
+        }
+
+        private void NotifyDescendantFlowDirection()
+        {
+            foreach (var visual in this.GetVisualDescendants())
+            {
+                if (visual is Control child)
+                {
+                    child.InvalidateFlowDirection();
+                }
+            }
+        }
+
+        private void AddMirrorTransform()
+        {
+            if (_hasMirrorTransform)
+            {
+                return;
+            }
+
+            var mirrorTransform = MirrorTrasform();
+            var rendertransform = RenderTransform;
+
+            ITransform? finalTransform = mirrorTransform;
+            if (rendertransform != null)
+            {
+                finalTransform = MargeTransforms(rendertransform, mirrorTransform);
+            }
+
+            base.RenderTransform = finalTransform;
+            _hasMirrorTransform = true;
+        }
+
+        private void RemoveMirrorTransform()
+        {
+            if (!_hasMirrorTransform)
+            {
+                return;
+            }
+
+            var mirrorTransform = MirrorTrasform();
+            var rendertransform = RenderTransform;
+            
+            ITransform? finalTransform = MargeTransforms(rendertransform, mirrorTransform);
+            if (finalTransform!.Value == Matrix.Identity)
+            {
+                finalTransform = null;
+            }
+
+            _hasMirrorTransform = false;
+            base.RenderTransform = finalTransform;
+        }
+
+
+        /// <summary>
+        /// Determines whether the element should be inverted if the 
+        /// flow direction is RightToLeft
+        /// </summary>
+        protected virtual bool ShouldGetInvertedIfRightToLeft() => true;
```

Comment:
I like WPF's naming here: `ShouldApplyMirrorTransform`

Reply:
It doesn't returns if a transform should be applied but whether control should be presented mirrored

Reply:
ShouldPresentedMirrored it's good?

Reply:
`IsMirrored`? `IsMirroredForRightToLeft`?

> ShouldPresentedMirrored it's good?

Correct English would be `ShouldPresentMirrored ` but I'm not a fan of the naming either.

Reply:
`IsMirrored` also does not seem so true to me, because not always when calling the method, the transform is applied (Even through the parent), it is also used to determine whether to apply a transform.

Reply:
The word `applied` is a bit problematic, as @Gillibald said, not always transform is applied, sometimes it is presented as a mirror because its visual father is presented as such.